device/common/include/traccc/device/make_sum_buffer.hpp

```diff
@@ -0,0 +1,59 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/device/fill_prefix_sum.hpp"
+#include "traccc/device/get_prefix_sum.hpp"
+#include "traccc/utils/memory_resource.hpp"
+
+// VecMem include(s).
+#include <vecmem/utils/copy.hpp>
+
+// System include(s).
+#include <variant>
+
+namespace traccc::device {
+
+/** @struct make_result_t
+ * @brief Helper class for return type of make_sum_buffer function
+ * @tparam T type of object within vector
+ * @param view View on the result vector
+ * @param variant Object posessing memory of the result vector
+ * @param totalSize Last member of result vector
+ */
+template <class T>
+struct make_buffer_result_t {
+    vecmem::data::vector_view<T> view;
+    std::variant<vecmem::vector<T>, vecmem::data::vector_buffer<T>> variant;
+    T totalSize;
+};  // struct make_buffer_result_t
+
+/// Function providing the prefix sum for a "size vector"
+///
+/// This simple function is just meant to translate the received "size vector"
+/// into a "prefix sum". I.e. into a list of index pairs that would allow
+/// visiting all elements of the jagged vector described by this "size vector".
+///
+/// @param sizes The sizes of the "inner vectors" of a jagged vector
+/// @param copy A "copy object" capable of dealing with the view
+/// @param mr    The memory resource to use of the result
+/// @return The prefix sum for efficiently iterating over all element of a
+///         jagged vector
+///
+TRACCC_HOST
+make_buffer_result_t<prefix_sum_size_t> make_sum_buffer(
+    const std::vector<prefix_sum_size_t> sizes,
```

Comment:
I didn't pass this as const reference because in seed_finding I'm calling this function directly with copy->get_sizes(...). If I were to pass by reference I'd need to in each call first create a 

```
auto foo = copy->get_sizes(...)
make_sum_buffer(foo, ...)
```

Reply:
Oh, of course. I was already doing it in fact in the previous version within the traccc::sycl::make_prefix_sum_buff. I just got a compiler error when making this change which led me to think about this the wrong way. It's fixed now